pandas/core/arrays/string_.py

```diff
@@ -382,6 +382,13 @@ def _values_for_factorize(self):
         arr[mask] = -1
         return arr, -1
 
+    @classmethod
+    def _from_factorized(cls, values, original):
+        assert values.dtype == original._ndarray.dtype
+        # When dropna (i.e. ignore_na) is False, can get -1 from nulls
```

Comment:
is there any way we could avoid this?

Reply:
Thanks - yes. Changed _values_for_factorize from -1 to None.